pkg/tbtc/chain.go

```diff
@@ -219,6 +219,21 @@ type BridgeChain interface {
 	) (*RedemptionRequest, error)
 }
 
+// NewWalletRegisteredEvent represents a new wallet registered event.
+type NewWalletRegisteredEvent struct {
+	EcdsaWalletID       [32]byte
+	WalletPublicKeyHash [20]byte
+	BlockNumber         uint64
+}
+
+// NewWalletRegisteredEventFilter is a component allowing to filter NewWalletRegisteredEvent.
+type NewWalletRegisteredEventFilter struct {
+	StartBlock          uint64
+	EndBlock            *uint64
+	EcdsaWalletID       [][32]byte
+	WalletPublicKeyHash [][20]byte
+}
```

Comment:
I need a clarification of this part of the PR description:

> We also moved the remaining chain types living so far in the `pkg/coordinator` (`coordinator. NewWalletRegisteredEvent` and `coordinator. NewWalletRegisteredEventFilter`) to the `pkg/tbtc` package. This package should be the right place for them according to the recent discussion (see https://github.com/keep-network/keep-core/pull/3650#discussion_r1242336911), at least until https://github.com/keep-network/keep-core/issues/3652 is done. 

This move makes sense if there are more than two chain interfaces using the type. In this particular case, only the `pkg/maintainer/wallet` uses it. Wouldn't it make more sense to keep this type defined inside the `pkg/maintainer/wallet/chain.go`?

Reply:
I think it will be much easier to declare all chain types in `pkg/tbtc`, regardless of what actually uses them. 

First reason: we avoid scattering those types around packages and we have everything in one place - the core `pkg/tbtc` package. This makes reasoning easier.

Second reason: Imagine `pkg/tbtc` needs to use those types one day. If so, we will have an import cycle and we will have to move those types to `pkg/tbtc` anyway. Also, if we introduce a new package `pkg/abc` that will need those types, it will be cleaner if it depends on `pkg/tbtc` and not on `pkg/maintainer/wallet`.

Reply:
That makes sense to me. Let's just make sure to be consistent about it cc @tomaszslabon @nkuba.